# Modelo Log-log ou Log-Linear

## Exemplo 1

Suponha que queiramos encontrar a elasticidade das despesas com bens
duráveis em relação às despesas totais de consumo pessoal.

**Significado das Variáveis:**

- DESPSERV: Despesas de serviços
- DESPDUR: Despesas com bens duráveis
- DESPNAODUR: Despesas não duráveis
- DESPTCP: Despesas totais de cosumo pessoal



In [12]:
#install.packages("gtsummary")
#install.packages("broom.helpers")

In [8]:
# Bibliotecas R
library(ggplot2)
library(readr)
library(gtsummary) # VIsualizar os resultados dos modelos


Leitura do dataset

In [5]:
df = read.csv( "https://raw.githubusercontent.com/pedroallmeiida/econometria/refs/heads/main/Dados/dados_exemplo_log_log.csv" )
df

DESPSERV DESPDUR DESPNAODUR DESPTCP
1  4143.3    971.4  2072.5     7184.9 
2  4161.3   1009.8  2084.2     7249.3 
3  4190.7   1049.6  2123.0     7352.9 
4  4220.2   1051.4  2132.5     7394.3 
5  4268.2   1067.0  2155.3     7479.8 
6  4308.4   1071.4  2164.3     7534.4 
7  4341.5   1093.9  2184.0     7607.1 
8  4377.4   1110.3  2213.1     7687.1 
9  4395.3   1116.8  2241.5     7739.4 
10 4420.0   1150.8  2268.4     7819.8 
11 4454.5   1175.9  2287.6     7895.3 
12 4476.7   1137.9  2309.6     7910.2 
13 4494.5   1190.5  2342.8     8003.8 
14 4535.4   1190.3  2351.1     8055.0 
15 4566.6   1208.8  2360.1     8111.2

Correlação

In [13]:
cor(df)

DESPSERV  DESPDUR   DESPNAODUR DESPTCP  
DESPSERV   1.0000000 0.9735536 0.9891647  0.9969194
DESPDUR    0.9735536 1.0000000 0.9770823  0.9855626
DESPNAODUR 0.9891647 0.9770823 1.0000000  0.9958636
DESPTCP    0.9969194 0.9855626 0.9958636  1.0000000

###  Modelo Log log para medir a Elasticidade da variável DESPDUR com DESPTCP

In [15]:
lm_loglog <- lm( log(DESPDUR)~ log(DESPTCP), data = df)
summary(lm_loglog)


Call:
lm(formula = log(DESPDUR) ~ log(DESPTCP), data = df)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0234363 -0.0037955  0.0008182  0.0066518  0.0163937 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  -7.54166    0.71615  -10.53 9.84e-08 ***
log(DESPTCP)  1.62661    0.08007   20.32 3.12e-11 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.01165 on 13 degrees of freedom
Multiple R-squared:  0.9695,	Adjusted R-squared:  0.9671 
F-statistic: 412.7 on 1 and 13 DF,  p-value: 3.118e-11


**Intrepretação do modelo log-log:**

A Equação do modelo predito é dada por:

$$
\widehat{\ln DESPDUR}_t = -7.54 + 1.63 \, \ln DESPTCP_t
$$

em que as estimativas para o intercepto e coeficiente de regressão são dados respectivamente por, $\hat\beta_0 = -7.54$ e $\hat\beta_1 = 1.63$.

A Elasticidade de DESPDUR em relação a DESPTCP é de cerca de 1,63, sugerindo que quando as despesas totais aumentam em 1%, as despesas com bens duráveis aumentam cerca de 1,63% em média.




In [16]:
#tbl_regression(lm_loglog, intercept = T)

###  Modelo Log log para medir a Elasticidade da variável DESPSERV com DESPTCP

In [17]:
lm_loglog <- lm( log(DESPSERV)~ log(DESPTCP), data = df)
summary(lm_loglog)


Call:
lm(formula = log(DESPSERV) ~ log(DESPTCP), data = df)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.004900 -0.001634  0.000090  0.002143  0.002998 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)   1.11614    0.15852   7.041 8.79e-06 ***
log(DESPTCP)  0.81203    0.01772  45.817 9.28e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.002579 on 13 degrees of freedom
Multiple R-squared:  0.9938,	Adjusted R-squared:  0.9934 
F-statistic:  2099 on 1 and 13 DF,  p-value: 9.282e-16


**Intrepretação do modelo log-log:**

A Equação do modelo predito é dada por:

$$
\widehat{\ln DESPSERV}_t = 1.12 + 0.81 \, \ln DESPTCP_t
$$

em que as estimativas para o intercepto e coeficiente de regressão são dados respectivamente por, $\hat\beta_0 = 1.12$ e $\hat\beta_1 = 0.81$.

A Elasticidade de DESPSERV em relação a DESPTCP é de cerca de 0.81, sugerindo que quando as despesas totais aumentam em 1%, as despesas com bens duráveis aumentam cerca de 0.81% em média.

Ou ainda, um aumento de 10% nas despesas de serviços, fazem com que as despesas com bens duráveis aumentam cerca de 8.1% em média.